archaius2-core/src/main/java/com/netflix/archaius/DefaultMethodInvokerFactory.java

```diff
@@ -0,0 +1,238 @@
+            INSTANCE = new LegacyJava8Factory();
+        } else if (SystemUtils.isJavaVersionAtMost(JavaVersion.JAVA_16)) {
+            INSTANCE = new BoundMethodHandlesFactory();
+        } else { // Java 17 onwards
+            INSTANCE = new LambdasFactory();
+        }
+        LOG.info("Choosing {} as invoker factory for default methods", INSTANCE.getClass());
+    }
+
+    static DefaultMethodInvokerFactory getFactory() {
+        return INSTANCE;
+    }
+
+    /**
+     * Returns a Function object that acts as a call to <code>invokerBound.method(...)</code>.
+     * <p>
+     * {@literal <}R> is expected to be the METHOD's return type. Unfortunately this constraint can't be enforced by the compiler.
+     *
+     * @param methodOwner An interface that declares (or inherits) the wanted method.
+     * @param method The method to be called. This is expected to be a <code>default</code> method in METHOD_OWNER.
+     * @param invokerBound An instance of the METHOD_OWNER interface.
+     */
+     abstract <T, R> Function<Object[], R> findOrCreateDefaultMethodInvoker(Class<T> methodOwner, Method method, T invokerBound);
+
+    /**
+     * Shared "fall back" implementation using reflection to invoke the method handle.
+     */
+     <T, R> Function<Object[], R> bindMethodHandle(MethodHandle methodHandle, T invokerBound) {
+         // Exceptions from this call are NOT caught because we want to fail fast if the arguments to the factory are bad.
+         MethodHandle boundHandle = methodHandle.bindTo(invokerBound);
+
+         return (args) -> {
+             try {
+                 if (boundHandle.type().parameterCount() == 0) {
+
+                     //noinspection unchecked
+                     return (R) boundHandle.invokeWithArguments();
+                 } else if (args != null) {
+
+                     //noinspection unchecked
+                     return (R) boundHandle.invokeWithArguments(args);
+                 } else {
+                     // This is a handle to a method WITH arguments, being called with none. This happens when
+                     // invokerBound.toString() is called on an object that has a parametrized property AND the interface
+                     // provides a default method for it. There's no good default to return here, so we'll just use null
+                     return null;
+                 }
+             } catch (Throwable e) {
+                 maybeWrapThenRethrow(e);
+                 return null; // Unreachable, but the compiler doesn't know
+             }
+         };
+
+     }
+
+
+    /**
+     * For Java 17 onwards, we wrap the method calls in lambdas and return that. This skips the reflection machinery and
+     * provides a noticeable performance boost. This implementation is known to fail in java 9 - 11. It *should* work
+     * from 12 or 13 onwards, but we have only tested it in 17 and 21.
+     */
+    private static class LambdasFactory extends DefaultMethodInvokerFactory {
```

Comment:
if package-scoped (final, too) you could make a matrix test and benchmarks, or even fuzzing to rely less on seen in usage stuff.